lib/msf/base/serializer/readable_text.rb

```diff
@@ -594,28 +606,39 @@ def self.dump_options(mod, indent = '', missing = false)
   # @param indent [String] the indentation to use.
   # @return [String] the string form of the information.
   def self.dump_advanced_options(mod, indent = '')
-    tbl = Rex::Text::Table.new(
-      'Indent'  => indent.length,
-      'Columns' =>
-        [
-          'Name',
-          'Current Setting',
-          'Required',
-          'Description'
-        ])
-
-    mod.options.sorted.each do |name, opt|
-      next unless opt.advanced?
-      next unless Msf::OptCondition.show_option(mod, opt)
-      if mod.datastore.is_a?(Msf::DataStoreWithFallbacks)
-        val = mod.datastore[name]
-      else
-        val = mod.datastore[name].nil? ? opt.default : mod.datastore[name]
+    options = mod.options.map { |name, option| option }
+    options_grouped_by_conditions = options.group_by do |option|
+      option.conditions
+    end
+    result = "".dup
+    options_grouped_by_conditions.each do |conditions, options|
+      tbl = Rex::Text::Table.new(
+        'Indent'  => indent.length,
+        'Columns' =>
+          [
+            'Name',
+            'Current Setting',
+            'Required',
+            'Description'
+          ])
+
+      options.sort_by(&:name).each do |opt|
+        next unless opt.advanced?
+        name = opt.name
+        if mod.datastore.is_a?(Msf::DataStoreWithFallbacks)
+          val = mod.datastore[name]
+        else
+          val = mod.datastore[name].nil? ? opt.default : mod.datastore[name]
+        end
+        tbl << [ opt.display_name || name, opt.display_value(val), opt.required? ? "yes" : "no", opt.desc ]
       end
-      tbl << [ name, opt.display_value(val), opt.required? ? "yes" : "no", opt.desc ]
+
+      next if conditions.any? && tbl.rows.empty?
+      result << "\n\n#{indent}When #{Msf::OptCondition.format_conditions(mod, options.first)}:\n\n" if conditions.any?
```

Comment:
Let's make this more clear to users by adding an extra word or two here

- Active when
- If ACTION = FOO:

Reply:
Addressed as part of #17526.